src/main/java/io/cdap/delta/transformation/MaskTransformation.java

```diff
@@ -49,30 +66,32 @@ private void parseDirective(TransformationContext context) {
       throw new IllegalArgumentException("Directive command line is null.");
     }
     String[] splits = commandLine.split(" ");
-    if (splits.length != 4) {
-      throw new IllegalArgumentException("Directive should have 4 arguments. Usage: mask column_name from_position " +
-                                           "to_position.");
+    if (splits.length != 5) {
+      throw new IllegalArgumentException("Directive is missing some parts or containing more parts," +
+                                           "Expected: mask column_name direction mask_char n, " +
+                                           "given directive: " + commandLine);
     }
     if (!NAME.equals(splits[0])) {
       throw new IllegalArgumentException("Directive is not a mask transformation. Usage: mask column_name " +
-                                           "from_position to_position.");
+                                           "direction mask_char n.");
     }
 
     srcColumn = splits[1];
-    try {
-      fromPos = Integer.parseInt(splits[2]);
-    } catch (NumberFormatException e) {
-      throw new IllegalArgumentException("from_position is not an integer.", e);
+    direction = splits[2];
+    if (!(direction.equals(RIGHT_DIRECTION) || direction.equals(LEFT_DIRECTION))) {
+      throw new IllegalArgumentException(String.format("masking direction should be %s or %s",
+                                                       RIGHT_DIRECTION, LEFT_DIRECTION));
+    }
+    
+    maskCharacter = splits[3];
+    if (maskCharacter.length() != 1) {
+      throw new IllegalArgumentException(String.format("masking_character: %s is not a character", maskCharacter));
     }
 
     try {
-      toPos = Integer.parseInt(splits[3]);
+      countN = Integer.parseInt(splits[4]);
```

Comment:
Should be whole number ( >=0 ) 

I think it should be negative. right?

Reply:
Good catch, added a check for this.